### PR TITLE
Implement caching, rename generated db entity model names 

### PR DESCRIPTION
**Pull request by mbrizic**
Includes:
- change generated db model names to be camelcase for consistency
- add "Rate events from this week" query on home page, also hidden the website description text if you're logged in
- implement caching of users, sports and events (approach is questionable and open for discussion, [description in readme](https://github.com/mbrizic/ratethegame/pull/39/files#diff-5a831ea67cf5cf8703b0de46901ab25bd191f56b320053be9332d9a3b0d01d15R28))
- change database primary keys to use identity columns instead of serial:
https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_serial
https://www.postgresqltutorial.com/postgresql-identity-column/

### CODE REVIEW

**Comment by mbrizic** [comment]
image

File: server/users/users.model.ts [hunk 1]
```diff
@@ -2,41 +2,31 @@ import { DomainModel } from "../core/domain.model"
...
 		id: number | undefined,
 		email: string,
-		password: string,
 		isAdmin: boolean,
 		subscriptions: UserSportSubscriptionsModel[],
 		settings: UserSettingsModel
 	) {
 		this.id = id
 		this.email = email
-		this.password = password
```
Comment (mbrizic) [inline server/users/users.model.ts:27]: @jurebb I removed this as I think it's safer we keep it out of domain model. Plus it's not really used right now, apart from length validation which is done in service anyway.

**Review by jurebb — COMMENTED** [review]
Overall, can't say I'm a huge fan of the current cache implementation (manual handling in services) -- but this is a great foundation and caching is almost certanly a "necessary evil" for this project.
I also think you've made some great points about caching implementation in the readme file.
I'd just like to add a few points

- caching could even be implemented on the frontend if we ever migrate to a fatter client/frontent model
- caching approaches should pose a few trade-offs: caching domain models may introduce too many invalid caches, while caching raw db models should introduce more required processing. Maybe one approach will suit our needs better if one of these trade-offs goes to its extreme. For this, we should monitor our analytics.
- In any case, even if we don't change our core models and services, I'd go towards eventually cleaning the implementation a bit -- some kind of automation or wrapping / attributed etc. This should ease maintainability/readability
- As you've (@mbrizic) pointed out offline, in case we transition to a db-model-based caching, this will probably entail a new layer (e.g. repository) which will include DB calls and caching logic

File: server/sports/sports.service.ts [hunk 2]
```diff
@@ -51,7 +70,7 @@ export default class SportsService {
 		const created = await Sports.create({
 			name: sport.name,
 			description: sport.description,
-			created_by: sport.createdByUserId
+			createdBy: sport.createdByUserId
 		})
```
[the remark this answers is not in the document]
Comment (mbrizic) [reply server/sports/sports.service.ts:75]: Good catch, I didn't even test this case. Perfect example of bugs we may have if we retain this approach (but not even db layer caching would help here).

File: server/users/users.service.ts [hunk 3]
```diff
@@ -12,33 +12,38 @@ import { UserSettings } from '../../database/models/user_settings';
...
+		const users = await Users.findAll({ include: entitiesToInclude });
 
-		return Promise.all(users.map(user => this.FromDatabaseWithSports(user)));
+		return users.map(user => UserFactory.FromDatabase(user, sports));
 	}
 
 	public async getById(userId: number): Promise<UserModel> {
-		const user = await Users.findByPk(userId, { include: this.entitiesToInclude });
+		const retrieved = usersCache.get(userId)
+		if (retrieved) {
```
[the remark this answers is not in the document]
Comment (mbrizic) [reply server/users/users.service.ts:33]: It works the same way, but I've changed it now everywhere to be less verbose.

File: server/admin/admin.controller.ts [hunk 4]
```diff
@@ -70,6 +72,16 @@ export class AdminController {
...
+	public clearEventsCache = async (req: RequestWithPotentialUser, res: Response, next: NextFunction) => {
+		clearEventsCaches()
+		res.redirect("/admin")
+	}
+
+	public clearSportsCache = async (req: RequestWithPotentialUser, res: Response, next: NextFunction) => {
+		clearSportsCaches()
+		res.redirect("/admin")
+	}
+
```
[the remark this answers is not in the document]
Comment (mbrizic) [reply server/admin/admin.controller.ts:85]: Fixed.

**Review by jurebb — APPROVED** [review]

**Pull request merged**